src/vos/evtree.c

```diff
@@ -3040,11 +3041,19 @@ void
 evt_desc_csum_fill(struct evt_context *tcx, struct evt_desc *desc,
 		   const struct evt_entry_in *ent)
 {
-	const daos_csum_buf_t *csum = &ent->ei_csum;
-	daos_size_t csum_buf_len = evt_csum_buf_len(tcx, &ent->ei_rect.rc_ex);
+	const daos_csum_buf_t	*csum = &ent->ei_csum;
+	daos_size_t		 csum_buf_len;
 
-	D_ASSERT(csum->cs_buf_len >= csum_buf_len);
-	memcpy(desc->pt_csum, csum->cs_csum, csum_buf_len);
+	if (!dcb_is_valid(csum))
+		return;
+
+	csum_buf_len = evt_csum_buf_len(tcx, &ent->ei_rect.rc_ex);
+	if (csum->cs_buf_len < csum_buf_len) {
```

Comment:
how could this happen? and what should we do for it?

Reply:
Don't remember exactly why I changed this. I think I just want to be cautious about checksum code asserting and causing the server to fail, especially because it's not a supported feature yet. Checksums are still part of the public API and therefore can get bad input, even if checksums are disabled. Currently, there aren't any guards in place to protect against that. Might be something I add in the next PR.